OpenIGTLinkIF/Logic/vtkSlicerOpenIGTLinkIFLogic.cxx

```diff
@@ -690,8 +715,87 @@ void vtkSlicerOpenIGTLinkIFLogic::GetDeviceNamesFromMrml(IGTLMrmlNodeListType &l
+    strcmp(transformNode->GetAttribute("IGTLVisible"), "true") == 0);
+  vtkMRMLModelNode* locatorModel = vtkMRMLModelNode::SafeDownCast(
+    transformNode->GetNodeReference(this->External->GetLocatorModelReferenceRole()));
+  if (!locatorModel && visible)
+    {
+    if (this->External->GetMRMLScene())
+      {
+      std::stringstream ss;
+      ss << "Locator_" << transformNode->GetName();
+      std::string locatorString = ss.str();
+      locatorModel = this->AddLocatorModel(this->External->GetMRMLScene(), ss.str(), 0.0, 1.0, 1.0);
+      if (locatorModel)
+        {
+        locatorModel->SetAndObserveTransformNodeID(transformNode->GetID());
+        transformNode->SetNodeReferenceID(this->External->GetLocatorModelReferenceRole(), locatorModel->GetID());
+        }
+      }
+    }
+  if (locatorModel)
+    {
+    locatorModel->SetDisplayVisibility(visible);
+    }
+}
+
+//---------------------------------------------------------------------------
+vtkMRMLModelNode* vtkSlicerOpenIGTLinkIFLogic::vtkInternal::AddLocatorModel(vtkMRMLScene * scene, std::string nodeName, double r, double g, double b)
+{
+
+  vtkSmartPointer<vtkMRMLModelNode> locatorModel = vtkSmartPointer<vtkMRMLModelNode>::New();
+  vtkSmartPointer<vtkMRMLModelDisplayNode> locatorDisp = vtkSmartPointer<vtkMRMLModelDisplayNode>::New();
+
+  // Cylinder represents the locator stick
+  vtkSmartPointer<vtkCylinderSource> cylinder = vtkSmartPointer<vtkCylinderSource>::New();
+  cylinder->SetRadius(1.5);
+  cylinder->SetHeight(100);
+  cylinder->SetCenter(0, 0, 0);
+  cylinder->Update();
+
+  // Rotate cylinder
+  vtkSmartPointer<vtkTransformPolyDataFilter> transformFilter = vtkSmartPointer<vtkTransformPolyDataFilter>::New();
+  vtkSmartPointer<vtkTransform> transform = vtkSmartPointer<vtkTransform>::New();
+  transform->RotateX(90.0);
+  transform->Translate(0.0, -50.0, 0.0);
+  transform->Update();
+  transformFilter->SetInputConnection(cylinder->GetOutputPort());
+  transformFilter->SetTransform(transform);
+  transformFilter->Update();
+
+  // Sphere represents the locator tip
+  vtkSmartPointer<vtkSphereSource> sphere = vtkSmartPointer<vtkSphereSource>::New();
+  sphere->SetRadius(3.0);
+  sphere->SetCenter(0, 0, 0);
+  sphere->Update();
+
+  vtkSmartPointer<vtkAppendPolyData> append = vtkSmartPointer<vtkAppendPolyData>::New();
+  append->AddInputConnection(sphere->GetOutputPort());
+  append->AddInputConnection(transformFilter->GetOutputPort());
+  append->Update();
+
+  locatorModel->SetAndObservePolyData(append->GetOutput());
+
+  double color[3];
+  color[0] = r;
+  color[1] = g;
+  color[2] = b;
+  locatorDisp->SetColor(color);
+
+  scene->SaveStateForUndo();
+  scene->AddNode(locatorDisp);
+  vtkMRMLNode* lm = scene->AddNode(locatorModel);
+  locatorDisp->SetScene(scene);
+  locatorModel->SetName(nodeName.c_str());
+  locatorModel->SetScene(scene);
+  locatorModel->SetAndObserveDisplayNodeID(locatorDisp->GetID());
+  locatorModel->SetHideFromEditors(0);
```

Comment:
this is probably not needed (and if it is then it has to be set before adding the node to the scene)